### ISSUE TITLE
Swarm STAMP runs: no best-model selection — final model is last round, not best

### ISSUE THREAD

**Issue by Ultimate-Storm** [KatherLab/MediSwarm#493]
## Summary

In **swarm** (`SwarmClientController` / CCWF) `STAMP_classification` runs, no best-model selection happens: the swarm best metric stays `None`, so the final distributed `FL_global_model.pt` is the **last round's** model, not the best-validation checkpoint.

## Evidence (client logs)

```
Gatherer - ... - global best metric is None from client None at round None
SwarmClientController - ... - Swarm Learning Done: number of rounds completed 20
SwarmClientController - ... - No global best result!
SwarmClientController - ... - broadcasting last result with metric None at round 19 to clients [...]
```

Because selection never triggers, whatever the **last** round produced is broadcast as the final model.

## Impact

In a 20-round run, one site's **final-round** validation loss spiked well above an earlier round (e.g. ~0.13 at round 17 → ~0.29 at round 19). With no best-model tracking, that late, worse model is the one kept and distributed — the better earlier checkpoint is lost.

## Root cause

The per-round client validation metric is not reported into the swarm controller's best-model tracker (`metric = None`), so intra-swarm model selection is never evaluated.

## Proposed fix

- Report the client validation metric (e.g. `val_loss` or `val_auroc`) to the swarm controller so best-model selection keeps the best checkpoint across rounds.
- Log/expose which round + metric value was selected as the final model.

## Related

- #476 (deploy-test evaluation is skipped)
- Companion issue: swarm-run eval metrics (AUROC) not retrievable centrally.

**Comment by Ultimate-Storm** [comment]
Companion: #492 (swarm eval metrics not retrievable centrally). Both surfaced together from the first full 20-round swarm run and relate to #476.

**Comment by Ultimate-Storm** [comment]
Fixed by #498 (merged to main, squash `8217ff6`). The client now logs `val_auroc` each round and `IntimeModelSelector` is keyed on `val_auroc`, so best-model selection fires and the best round's model is kept/broadcast instead of the last. (It stayed open because the PR body used "Fixes #492 and #493" — a single `fixes` keyword only auto-closes the first issue.) Note: verified by unit tests + CI, but the swarm-metric flow still needs a post-rebuild 2-site smoke to confirm end-to-end.

**Comment by Ultimate-Storm** [comment]
Reopening — I closed this prematurely. The 2-node validation run on the built image (`1.6.0-dev.260729.4b9deba`, job `ccdb342c`) shows best-model selection still does **not** fire:

```
Gatherer - global best metric is None from client None at round None
SwarmClientController - No global best result!
```

**Root cause (now established from the run, not inferred).** `SwarmClientController.best_metric` is set only when `IntimeModelSelector` fires `GLOBAL_BEST_MODEL_AVAILABLE`, which requires `key_metric` to be present in the metrics submitted with the result. The selector logged exactly what it received:

```
IntimeModelSelector - WARNING - validation metric `val_auroc` not in metrics from Mainz_1: ['validation_loss', 'validation_auroc']
```

So the metric does flow — under the key **`validation_auroc`**. #498 set `key_metric = "val_auroc"`, which never matches, so selection never fires and the last round's model is still what gets broadcast.

**Fix:** `key_metric = "validation_auroc"`. Also worth adding `validation_auroc` to the fallback key list in `stamp_metrics_callback`, since that is the name STAMP actually logs.

Lesson for the record: unit tests + CI could not catch this — only a real swarm run surfaces which metric keys reach the selector.
